File: src/backend/app/Models/Room.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Thiagoprz\CompositeKey\HasCompositeKey;

class Room extends Model
{
    use HasCompositeKey;

    protected $table = 'room';

    protected $primaryKey = ['lodgingId', 'roomNumber'];
    
    public $timestamps = false;

    protected $fillable = [
        'lodgingId',
        'roomNumber',
        'occupied',
        'perNightPrice',
        'capacity'
    ];

    public function lodging()
    {
        return $this->belongsTo(Lodging::class);
    }
}

File: src/backend/src/api/Models/Room.cs
namespace Restify.API.Models;

public class Room
{
	public int	LodgingId { get; set; }
	public int	Number { get; set; }
	public int	TypeId { get; set; }

	public Lodging	Lodging { get; }
	public RoomType	Type { get; set; }
	
	public Room() {}

	public Room(RoomType type)
	{
		 Type = type;
	}
}